### ISSUE TITLE
Incorrect indentation for brace-style block with multi-line expression

### ISSUE THREAD

**Issue by owst** [vim-ruby/vim-ruby#314]
Using `119967a` the following source code:

```
def foo(param_a)
  variable_a, variable_b = some_unreasonably_long_object_name.some_long_method_name(param_a) { |block_arg|
    true
  }
end
```

is indented fine, but when I try and fit the method into 100 characters width, as:

```
def foo(param_a)
  variable_a, variable_b = 
    some_unreasonably_long_object_name.some_long_method_name(param_a) { |block_arg|
      true
  }
end
```

the indentation is incorrect - the closing `}` should be aligned with the `s` of `some_unreasonably_long_object_name`, rather than the `v` of `variable_a`.

N.B. with `do...end` style blocks, the indentation _is_ correct:

```
def foo(param_a)
  variable_a, variable_b = 
    some_unreasonably_long_object_name.some_long_method_name(param_a) do |block_arg|
      true
    end
end
```

**Comment by woahdae** [comment]
The offending line is [here](https://github.com/vim-ruby/vim-ruby/blob/416be2b00de6390f05da2f60f18aa15a4397eef7/indent/ruby.vim#L328):

```vimscript
      if closing_bracket == ')' && col('.') != col('$') - 1
        let ind = virtcol('.') - 1
      elseif g:ruby_indent_block_style == 'do'
        let ind = indent(line('.'))
      else " g:ruby_indent_block_style == 'expression'
        " *** Removing this condition and changing the above to 'else', i.e. treating blocks and expressions the same, fixes the issue. 
        " *** Question is, why use GetMSL?
        let ind = indent(s:GetMSL(line('.')))
      endif
```

As I noted in the comment above, treating both blocks and expressions with `indent(line('.'))` instead of using `s:GetMSL` for expressions fixes the issue.

Obviously, someone put GetMSL there for a reason, but I can't figure out what it is. I'll make the change locally and see if I can catch what breaks, if anything.

Maybe it's just a different syntax preference? In which case we could add a configuration for the preference?

**Comment by woahdae** [comment]
I think this can be resolved. Related: https://github.com/vim-ruby/vim-ruby/issues/328

TL;DR set `let g:ruby_indent_block_style = 'do'`

The [documentation](https://github.com/vim-ruby/vim-ruby/blob/1b4a95db4931317a34fb55091624a4f976728264/doc/vim-ruby.txt#L116) (not sure how to get to that via `:help`)

I didn't figure this out until I dug in to the code, not sure how to make it more obvious.

**Comment by AndrewRadev** [comment]
Thanks for the help, @woahdae.

This is a somewhat common problem, and I'm thinking the "do" indent block style (a terrible set of names for this, I know) might be a better default. I've pushed a commit that makes it the default, now let's see if we start getting bug reports for the opposite formatting :). With this, I'll close the issue for the moment.

> I didn't figure this out until I dug in to the code, not sure how to make it more obvious.

I'll do some thinking on how to make the documentation a bit easier to access.

**Comment by owst** [comment]
Thanks @woahdae 👍

**Comment by AndrewRadev** [comment]
Update: I've updated the README with some more information on where to find settings and such. I've also added `vim-ruby-` help tags, so that attempting to type `:help vim-ruby<tab>` would autocomplete to something sensible. I'm hoping this will improve discoverability of settings a bit.
